### PR TITLE
Close #157 splash2xdmf iteration

### LINKED ISSUE #157: Iteration instead of time step
Make sure we always refer to "step" instead of "timestep" or similar.

A "step" does not need to be related to time, even if it is in PIConGPU (right now).

(To add more confusion: in a [boosted frame simulation](http://dx.doi.org/10.1016/j.jcp.2011.04.003), a step does not necessarily mean that all the data of this step is actually "measured" at the *same time*.)

### PR DESCRIPTION
**Pull request by ax3l**
Follow up to #187 to close #157: Draws the connection between libSplash iterations and XDMF time steps in splash2xdmf.py

to do:
- [x] run time test
- [x] cross check against [pic2xdmf.py](https://github.com/ComputationalRadiationPhysics/picongpu/blob/ac82b4c96d5659380042b0e61f14092961e36120/src/tools/bin/pic2xdmf.py) to see if all hooks are still working correctly (from first view: not affected, run time test: pending)

### CODE REVIEW

File: tools/splash2xdmf.py [hunk 1]
```diff
@@ -411,9 +411,10 @@ def parse_hdf5_recursive(h5Group, h5filename, level):
             print_dataset(h5Group[g], level, h5filename)
     
 
-def get_timestep_for_group(dataGroup):
+def get_iteration_for_group(dataGroup):
```
Comment (ax3l) [inline tools/splash2xdmf.py]: API change

**Comment by f-schmitt-zih** [comment]
@ax3l do you still plan to take up those todo's?

**Comment by ax3l** [comment]
@f-schmitt-zih yes, sry :) I think they were fine but I will re-check

**Comment by ax3l** [comment]
@f-schmitt-zih checked :)

**Pull request merged**